From 15adcb458c697f8e1aed84de43196cea3a2eae97 Mon Sep 17 00:00:00 2001
From: Niels de Vos <ndevos@redhat.com>
Date: Wed, 23 Nov 2016 11:31:15 +0100
Subject: [PATCH] FSAL_VFS: initialize 'unix_mode' to prevent compiler error

CentOS-6 builds complain that 'unix_mode' could be used uninitialized.
Inspecting the code, this does not look possible, but the (old) compiler
is not smart enough.

This change addresses the following compile error:

  nfs-ganesha-2.4.1/src/FSAL/FSAL_VFS/file.c:330: error: 'unix_mode' may be used uninitialized in this function

Signed-off-by: Niels de Vos <ndevos@redhat.com>
---
 src/FSAL/FSAL_VFS/file.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/FSAL/FSAL_VFS/file.c b/src/FSAL/FSAL_VFS/file.c
index 1192be4..0c4d9a2 100644
--- a/src/FSAL/FSAL_VFS/file.c
+++ b/src/FSAL/FSAL_VFS/file.c
@@ -327,7 +327,7 @@ fsal_status_t vfs_open2(struct fsal_obj_handle *obj_hdl,
 	int posix_flags = 0;
 	int fd, dir_fd;
 	int retval = 0;
-	mode_t unix_mode;
+	mode_t unix_mode = 0000; /* gets set below, old compilers complain */
 	fsal_status_t status = {0, 0};
 	struct vfs_fd *my_fd = NULL;
 	struct vfs_fsal_obj_handle *myself, *hdl = NULL;
-- 
2.7.4

